sale_blanket_order/views/sale_blanket_order_views.xml

```diff
@@ -117,7 +121,7 @@
                         <page string="Order Lines" name="order_lines">
                             <field
                                 name="line_ids"
-                                attrs="{'readonly': [('state', 'in', ('open','expired'))]}"
+                                attrs="{'readonly': [('state', 'in', ('open','cancel'))]}"
```

Comment:
**issue**: if you leave both states ('cancel' and 'expired' then this `attrs` and other attributes and functions in the module need to be updated accordingly.